### PR TITLE
Terminal state logic

### PR DESCRIPTION
**Pull request by ben-bay**
Improved terminal state logic. Other ideas? What do you think?

Going off ideas from #28 by @Akababa

### CODE REVIEW

**Comment by nickwalton** [comment]
Let's try it out! Maybe we should start developing a set of test cases of things that are clearly terminal cases and things that are easy to mistake for terminal ones, but really aren't? Though that might be a larger task

**Comment by ben-bay** [comment]
> Let's try it out! Maybe we should start developing a set of test cases of things that are clearly terminal cases and things that are easy to mistake for terminal ones, but really aren't? Though that might be a larger task

Good idea, will set something up for that.

File: story/utils.py [hunk 1]
```diff
@@ -47,41 +47,30 @@ def get_num_options(num):
...
+    TODO: Add in more sophisticated NLP, maybe a custom classifier
+    trained on hand-labelled data that classifies second-person
+    statements as resulting in death or not.
+    """
+    lower_text = text.lower()
+    you_dead_regexps = [
+        "you('re| are) (dead|killed|slain|no more|nonexistent)",
+        "you (die|pass away|perish|suffocate|drown|bleed out)",
+        "you('ve| have) (died|perished|suffocated|drowned|been (killed|slain))",
+        "you \w* to death",
```
Comment (Akababa) [inline story/utils.py]: can be combined into a single regex `you \w+( yourself) to death`

File: story/utils.py [hunk 2]
```diff
@@ -47,41 +47,30 @@ def get_num_options(num):
...
-
-    won_phrases = ["live happily ever after", "you live forever"]
-    for phrase in won_phrases:
-        if phrase in text:
-            return True
-    return False
+    lower_text = text.lower()
+    won_phrases = [
+        "live happily ever after",
+        "(you)? live (forever|eternally|for eternity)",
```
Comment (Akababa) [inline story/utils.py]: Could be too general, matches for example: `god will live forever`

Comment (ben-bay) [reply story/utils.py]: Thanks fixed

File: story/utils.py (hunk 1 above)
Comment (ben-bay) [reply story/utils.py]: @Akababa  good point! I also wanted to allow for: `the ogre clubs you to death`. So maybe just `you \w*( yourself) to death`??

**Pull request merged**